### PR TITLE
Add edit/update support for calendar events

### PR DESCRIPTION
**Pull request by allenporter**
## Proposed change

Add edit/update support for calendar events. Some of the scaffolding was already created, and this fills it in with the websocket added to core in https://github.com/home-assistant/core/pull/84141

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/calendar/dialog-calendar-event-editor.ts [hunk 1]
```diff
@@ -158,6 +160,7 @@ class DialogCalendarEventEditor extends LitElement {
             class="summary"
             name="summary"
             .label=${this.hass.localize("ui.components.calendar.event.summary")}
+            value=${this._summary}
```
Comment (bramkragten) [inline src/panels/calendar/dialog-calendar-event-editor.ts:163]:
```suggestion
            .value=${this._summary}
```

File: src/panels/calendar/dialog-calendar-event-editor.ts [hunk 2]
```diff
@@ -169,6 +172,7 @@ class DialogCalendarEventEditor extends LitElement {
             .label=${this.hass.localize(
               "ui.components.calendar.event.description"
             )}
+            value=${this._description}
```
Comment (bramkragten) [inline src/panels/calendar/dialog-calendar-event-editor.ts:175]:
```suggestion
            .value=${this._description}
```

File: src/panels/calendar/dialog-calendar-event-editor.ts [hunk 3]
```diff
@@ -457,7 +470,61 @@ class DialogCalendarEventEditor extends LitElement {
...
+        ),
+        confirmText: this.hass.localize(
+          "ui.components.calendar.event.confirm_update.update_this"
+        ),
+        confirmFutureText: this.hass.localize(
+          "ui.components.calendar.event.confirm_update.update_future"
+        ),
+      });
+    }
+    if (range === undefined) {
```
Comment (bramkragten) [inline src/panels/calendar/dialog-calendar-event-editor.ts:506]: This means we also never save an event that has not recurrence?

**Review by bramkragten — DISMISSED** [review]

**Review by bramkragten — APPROVED** [review]

**Comment by MartinHjelmare** [comment]
Backend PR is approved.

**Comment by allenporter** [comment]
Backend PR merged

**Comment by allenporter** [comment]
Merged backend PR

**Pull request merged**